src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/custom.py

```diff
@@ -107,54 +109,49 @@ def _handle_global_configuration(config):
+                                                                              'core', 'output',
+                                                                              OUTPUT_LIST))
+            answers['output_type_prompt'] = output_index
+            answers['output_type_options'] = str(OUTPUT_LIST)
+            enable_file_logging = prompt_y_n(MSG_PROMPT_FILE_LOGGING, default='n')
+            allow_telemetry = prompt_y_n(MSG_PROMPT_TELEMETRY, default='y')
+            answers['telemetry_prompt'] = allow_telemetry
+            cache_ttl = None
+            while not cache_ttl:
+                try:
+                    cache_ttl = prompt(MSG_PROMPT_CACHE_TTL) or DEFAULT_CACHE_TTL
+                    # ensure valid int by casting
+                    cache_value = int(cache_ttl)
+                    if cache_value < 1:
+                        raise ValueError
+                except ValueError:
+                    logger.error('TTL must be a positive integer')
+                    cache_ttl = None
+            # save the global config
+            config.set_value('core', 'output', OUTPUT_LIST[output_index]['name'])
+            config.set_value('core', 'collect_telemetry', 'yes' if allow_telemetry else 'no')
+            config.set_value('core', 'cache_ttl', cache_ttl)
+            config.set_value('logging', 'enable_log_file', 'yes' if enable_file_logging else 'no')
 
 
 # pylint: disable=inconsistent-return-statements
-def handle_configure(cmd, defaults=None, list_defaults=None):
+def handle_configure(cmd, defaults=None, list_defaults=None, scope=None):
     if defaults:
         defaults_section = cmd.cli_ctx.config.defaults_section_name
-        for default in defaults:
-            parts = default.split('=', 1)
-            if len(parts) == 1:
-                raise CLIError('usage error: --defaults STRING=STRING STRING=STRING ...')
-            cmd.cli_ctx.config.set_value(defaults_section, parts[0], _normalize_config_value(parts[1]))
+        with ConfiguredDefaultSetter(cmd.cli_ctx.config, scope.lower() == 'local'):
+            for default in defaults:
+                parts = default.split('=', 1)
+                if len(parts) == 1:
+                    raise CLIError('usage error: --defaults STRING=STRING STRING=STRING ...')
+                cmd.cli_ctx.config.set_value(defaults_section, parts[0], _normalize_config_value(parts[1]))
         return
     if list_defaults:
-        use_local_config_original = cmd.cli_ctx.config.use_local_config
-        cmd.cli_ctx.config.set_to_use_local_config(list_defaults == 'local')
-        defaults_result = cmd.cli_ctx.config.items(cmd.cli_ctx.config.defaults_section_name)
-        cmd.cli_ctx.config.set_to_use_local_config(use_local_config_original)
-        return defaults_result
+        with ConfiguredDefaultSetter(cmd.cli_ctx.config, scope.lower() == 'local'):
+            defaults_result = cmd.cli_ctx.config.items(cmd.cli_ctx.config.defaults_section_name)
```

Comment:
Why not just:
return cmd.cli_ctx.config.items(cmd.cli_ctx.config.defaults_section_name)

Reply:
we would like only display  valid defaults which have to have values

Reply:
Ok, good to know we have some keys that didn't have any values.